src/main/scala/org/broadinstitute/dsde/firecloud/service/StatusService.scala

```diff
@@ -1,43 +1,62 @@
 package org.broadinstitute.dsde.firecloud.service
 
-import java.text.SimpleDateFormat
-import java.util.Date
-
 import akka.actor.Actor
-import spray.json._
+import akka.actor.Props
+import akka.pattern._
+import org.broadinstitute.dsde.firecloud.Application
+import org.broadinstitute.dsde.firecloud.dataaccess.{AgoraDAO, RawlsDAO, SearchDAO, ThurloeDAO}
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol.impSystemStatus
+import org.broadinstitute.dsde.firecloud.model.{SubsystemStatus, SystemStatus}
+import org.broadinstitute.dsde.firecloud.service.PerRequest.{PerRequestMessage, RequestComplete}
+import org.broadinstitute.dsde.firecloud.service.StatusService.CollectStatusInfo
+import spray.http.StatusCodes
+import spray.httpx.SprayJsonSupport
+import spray.json.DefaultJsonProtocol._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext
+
+/**
+ * Created by anichols on 4/5/17.
+ */
+object StatusService {
+  def props(statusServiceConstructor: () => StatusService): Props = Props(statusServiceConstructor())
+
+  def constructor(app: Application)()(implicit executionContext: ExecutionContext): StatusService = {
+    new StatusService(app.searchDAO, app.agoraDAO, app.thurloeDAO, app.rawlsDAO)
+  }
+
+  case class CollectStatusInfo()
+}
 
-import spray.routing._
+class StatusService (val searchDAO: SearchDAO,
+                     val agoraDAO: AgoraDAO,
+                     val thurloeDAO: ThurloeDAO,
+                     val rawlsDAO: RawlsDAO)
+                    (implicit protected val executionContext: ExecutionContext) extends Actor with SprayJsonSupport {
 
+  def collectStatusInfo(): Future[PerRequestMessage] = {
+    val subsystemExceptionHandler: PartialFunction[Throwable, SubsystemStatus] = {
+      case e: Exception => SubsystemStatus(false, Some(Array(e.getMessage)))
+    }
 
-import org.broadinstitute.dsde.firecloud.FireCloudConfig
+    for {
+      rawlsStatus <- rawlsDAO.status recover subsystemExceptionHandler
+      thurloeStatus <- thurloeDAO.status recover subsystemExceptionHandler
+      agoraStatus <- agoraDAO.status recover subsystemExceptionHandler
+      searchStatus <- searchDAO.status recover subsystemExceptionHandler
+    } yield {
+      val statusMap = Map("Rawls" -> rawlsStatus, "Thurloe" -> thurloeStatus, "Agora" -> agoraStatus, "Search" -> searchStatus)
```

Comment:
Instead of hardcoding names here, let's have each DAO define the service name. So, in RawlsDAO.scala:
```
object RawlsDAO {
 ...
  lazy val serviceName = "Rawls"
  ...
}

trait RawlsDAO extends LazyLogging with ReportsSubsystemStatus {

  implicit val errorReportSource = ErrorReportSource(RawlsDAO.serviceName)
```

and here:
```
val statusMap = Map("Rawls" -> rawlsStatus, ...
```